### ISSUE TITLE
Update TriBITS GitHub Actions builds 2022-12

### ISSUE THREAD

**Issue by bartlettroscoe** [TriBITSPub/TriBITS#543]
### Description

The current set of TriBITS GitHub Actions builds are:

* **1)** CMake 3.17.5, makefiles, Python 2.7, gcc-8, g++8, gfortran-8
* **2)** CMake 3.17.5, makefiles, Python 3.8, gcc-10, g++10, gfortran-10
* **3)** CMake 3.17.5, makefiles, Python 3.8, gcc-11, g++11,
* **4)** CMake 3.21.2, makefiles, Python 3.8, gcc-9 g++9, gfortran-9, no_have_ninja: no-ninja
* **5)** CMake 3.23.1, makefiles, Python 3.8, gcc-9, g++9, gfortran-9

The TriBITS GitHub Actions builds need to be updated for the following reasons:

* **a)** A shown [here](https://github.com/TriBITSPub/TriBITS/actions/workflows/tribits_testing.yml), the `gcc-8` build starting failing on Friday, 12/2/2022 by failing to install the `gcc-8` package (see error message [here](https://github.com/TriBITSPub/TriBITS/actions/runs/3599448970/jobs/6063096593)).

* **b)** TriBITS will be upgrading the minimum version of CMake to 3.22.0 (see https://github.com/trilinos/Trilinos/issues/10355) (and all Trilinos PR and MM testing is using CMake 3.22+)

* **c)** TriBITS should be tracking the newest CMake version when it comes out

* **d)** There are newer versions of gcc that can and should be tested (GCC 12)

Note that we might also consider dropping testing for Python 2.7 but I think unless we loose support for installing Python 2.7 and as long as RHEL7 exists at SNL, it seems smart to keep Python 2.7 support.

### Initial Proposed Updated GitHub Actions Builds

Considering all of this, here are the updated proposed set of builds

* **1)** CMake 3.22.4, makefiles, Python 2.7, gcc-9, g++9, gfortran-9
* **2)** CMake 3.22.4, makefiles, Python 3.8, gcc-10, g++10, gfortran-10
* **3)** CMake 3.22.4, makefiles, Python 3.8, gcc-11, g++11
* **4)** CMake 3.23.1, makefiles, Python 3.8, gcc-9, g++9, gfortran-9, no_have_ninja: no-ninja
* **5)** CMake 3.25.1, makefiles, Python 3.8, gcc-12, g++12 gfortran-12

That should cover things pretty well.

### Proposed Updated GitHub Actions Builds 2

Given that Python 2.7 seems to no longer be supported, we can drop that build to give:

* **1)** CMake 3.22.4, makefiles, Python 3.8, gcc-10, g++10, gfortran-10
* **2)** CMake 3.22.4, makefiles, Python 3.8, gcc-11, g++11
* **3)** CMake 3.23.1, makefiles, Python 3.8, gcc-9, g++9, gfortran-9, no_have_ninja: no-ninja
* **4)** CMake 3.25.1, makefiles, Python 3.8, gcc-12, g++12 gfortran-12

### Tasks

* [ ] Create branch and PR (#544) with updated set of builds and get to pass ... **IN PROGRESS ...**
* [ ] Change the set of required builds to match new set of required builds 
* [ ] Merge PR

**Comment by bartlettroscoe** [comment]
Well, it looks like Python 2.7 is no longer available as shown by the new build [???](https://github.com/TriBITSPub/TriBITS/actions/runs/3621774657/jobs/6105637961) showing:

```
Run actions/setup-python@v2
Version 2.7 was not found in the local cache
Error: Version 2.7 with arch x[6](https://github.com/TriBITSPub/TriBITS/actions/runs/3621774657/jobs/6105637961#step:7:7)4 not found
The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
``` 

Therefore, I will need to drop the Python 2.7 build.  So it looks like we no longer have testing with Python 2.7 on GitHub Actions!

**Comment by bartlettroscoe** [comment]
The recently tested and merged PRs #546 and #549 show the updated GitHub Actions builds are working as they should.  Closing this Issue as complete.
